imvertorlite.py

```diff
@@ -1,137 +1,189 @@
-
-    # Convert the schema to JSON.
-    output = json.dumps(schema, indent=2)
-    with open(f"{schema['title'].lower().replace("/", "-")}.json", "w") as f:
-        f.write(output)
-
-
-@click.command()
-@click.option(
-    "file_name",
-    "--file",
-    "-f",
-    type=click.File("r"),
-    required=True,
-    help="An XML export from Enterprise Architect.",
-)
-@click.option(
-    "class_name", "--name", "-n", help="The UML class name of the object to export."
-)
-def convert(file_name, class_name=None):
-    """
-    Exports one or more UML classes from an Enterprise Architect XML export to 
-    JSON Schema.
-    """
-    res = etree.parse(file_name)
-
-    if class_name:
-        selector = f'[@name="{class_name}"]'
-    else:
-        # TODO: Filter out usefull classes.
-        selector = ""
-
-    clazzes = res.xpath(f"//UML:Class{selector}", namespaces=NAMESPACES)
-
-    for clazz in clazzes:
-        convert_clazz(res, clazz)
-
-
-if __name__ == "__main__":
-    convert()
+import json
+import sys
+import click
+from bs4 import BeautifulSoup, ResultSet, Tag
+from utils import type_conversion, type_convert_dictionary
+import cchardet
+
+
+def get_initial_value(element: Tag):
+    initial = element.select_one("initial").get("body")
+    if initial:
+        return initial
+    else:
+        return None
+
+
+def extract_properties(attr: Tag):
+    attr_name = attr.get("name")
+    attr_dict = {
+        "$id": f"#/properties/{attr_name}",
+        "title": f"{attr_name}",
+    }
+
+    attr_props = attr.select_one("properties")
+    attr_dict.update(
+        type_convert_dictionary(attr_props.attrs))
+
+    initial_value = get_initial_value(attr)
+    if initial_value:
+        attr_dict["default"] = type_conversion(initial_value)
+
+    attr_type = attr_props.get("type")
+    if attr_type:
+        attr_dict["type"] = type_conversion(attr_type)
+
+    stereotype = attr.select_one("stereotype")
+    if stereotype:
+        attr_dict["stereotype"] = stereotype.get("stereotype")
+
+    documentation = attr.select_one("documentation")
+    if documentation:
+        attr_dict["description"] = documentation.get("value")
+
+    tags = attr.select_one("tags").select("tag")
+    if tags:
+        attr_dict["tags"] = [{"name": tag.get("name"),
+                              "value": tag.get("value"),
+                              "notes": tag.get("notes")} for tag in tags]
+
+    return attr_dict
+
+
+def generate_schema(soup: BeautifulSoup, base: BeautifulSoup, attrs: ResultSet):
+    """
+    Converts a specific UML class to a JSON schema.
+
+    @param soup: The base soup of the XML file to resolve references.
+    @param base: The base UML class (BeautifulSoup object).
+    @param attrs: A ResultSet (list of elements) containing all the individual properties.
+    """
+
+    # Create a base schema.
+    schema = {
+        "$schema": "http://json-schema.org/draft-07/schema",
+        "$id": "https://objecttypes.vng.cloud/schema.json",

```

Comment:
```suggestion
        f"$id": "https://github.com/open-objecten/objecttypes/{base.get("name").lower()}.json",
```

Since we now have Open Objecten as a place to add these. This is prolly a better "id".